hash.c

```diff
@@ -1698,22 +1699,38 @@ tbl_update_modify(st_data_t *key, st_data_t *val, st_data_t arg, int existing)
     return ret;
 }
 
+static int
+tbl_update2(struct update_arg *update_info);
```

Comment:
Rearranging `tbl2_update` before `tbl_update` caused a lot of diff churn, which made the PR difficult to read.

I added this prototype for now, just to make reviewing easier.